pennylane/devices/default_mixed.py

```diff
@@ -0,0 +1,153 @@
+    #     "PhaseShift",
+    #     "RX",
+    #     "RY",
+    #     "RZ",
+    #     "Rot",
+    #     "CRX",
+    #     "CRY",
+    #     "CRZ",
+    #     "CRot",
+    #     "AmplitudeDamping",
+    #     "GeneralizedAmplitudeDamping",
+    #     "PhaseDamping",
+    #     "DepolarizingChannel",
+    #     "QubitChannel",
+    # }
+
+    def __init__(self, wires, *, shots=1000, analytic=True):
+        # call QubitDevice init
+        super().__init__(wires, shots, analytic)
+
+        # Create the initial state.
+        self._state = self._create_basis_state(0)
+        self._pre_rotated_state = self._state
+
+    def _create_basis_state(self, index):
+        """Return the density matrix representing a computational basis state over all wires.
+
+        Args:
+            index (int): integer representing the computational basis state.
+
+        Returns:
+            array[complex]: complex array of shape ``[2] * (2 * num_wires)``
+            representing the density matrix of the basis state.
+        """
+        rho = np.zeros((2 ** self.num_wires, 2 ** self.num_wires), dtype=np.complex128)
+        rho[index, index] = 1
+        rho = self._asarray(rho, dtype=self.C_DTYPE)
+        return self._reshape(rho, [2] * (2 * self.num_wires))
+
+    @property
+    def state(self):
+        """Returns the state density matrix of the circuit prior to measurement"""
+        dim = 2 ** self.num_wires
+        # User obtains state as a matrix
+        return self._reshape(self._pre_rotated_state, (dim, dim))
+
+    def reset(self):
+        """Resets the device"""
+        super().reset()
+
+        self._state = self._create_basis_state(0)
+        self._pre_rotated_state = self._state
+
+    def analytic_probability(self, wires=None):
+
+        if self._state is None:
+            return None
+
+        # convert rho from tensor to matrix
+        rho = self._reshape(self._state, (2 ** self.num_wires, 2 ** self.num_wires))
+        # probs are diagonal elements
+        probs = self.marginal_prob(self._diag(rho), wires)
+        return probs
+
+    def _get_kraus(self, operation):  # pylint: disable=no-self-use
+        """Return the Kraus operators representing the operation.
+
+        Args:
+            operation (~.Operation): a PennyLane operation
+
+        Returns:
+            list[array[complex]]: Returns a list of 2D matrices representing the Kraus operators. If
+            the operation is unitary, returns a single Kraus operator. In the case of a diagonal
+            unitary, returns a 1D array representing the matrix diagonal.
+        """
+        if isinstance(operation, DiagonalOperation):
+            return operation.eigvals
+
+        if isinstance(operation, Channel):
+            return operation.kraus_matrices
+
+        return [operation.matrix]
+
+    # Note: placeholder to be able to run tests
+    # pylint:disable=pointless-statement
+    def apply(self, operations, **kwargs):
+
+        lambda x: True
```

Comment:
```suggestion
        return True
```
This shouldn't make any difference either way though :laughing:

Reply:
If the function is not needed to be run (only needs to be declared), the usual way to indicate this is with `pass`

Reply:
Or even just a docstring counts, a `pass` isn't even needed!
```python
def apply(self, operations, **kwargs):
    """A dummy apply statement, to allow the device to be instantiated"""
```